sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -599,11 +596,9 @@ case class DropTable(
 case class AlterTable(
     catalog: TableCatalog,
     ident: Identifier,
-    table: NamedRelation,
+    table: Table,
```

Comment:
This was previously `NamedRelation` so that it could be converted from a statement when the catalog is resolved using `UnresolvedRelation`. Then the normal table resolution rule would run to resolve this as to a DSv2 relation. Why change that?

Reply:
As I think about this more, it seems like a bad idea.

This ties the SQL logical plans to a particular type of relation. There is no need to do that. What if we decide to add a v3 connector API? Then we would have to change all of these logical plans instead of using a new type of relation.

I think we should move this back to `NamedRelation` to keep logical plans and the connector API separate. That also makes rules cleaner, because catalog resolution and table resolution are done in independent rules.